client/components/command-pallette/README.md

```diff
@@ -0,0 +1,68 @@
+# Command Palette
+
+The Command Palette is a React component used to display a text input used to search and filter a list of commands. It can be accessed in the Calypso using the shortcut `cmd+k`.
+
+The command palette is inspired by [Core](https://github.com/WordPress/gutenberg/blob/trunk/packages/commands/README.md) implementation used in the Gutenberg editor. We use the same CSS and similar markup, the commands share the same basic props that we extended to support the following differences:
+
+- The Calypso version supports multisite commands with the sites as second step commands.
+- Users can go back to root commands with the `esc` and `backspace` keys, or the back button.
+- We change the search placeholder to inform the user about the current command.
+- Besides of icons, we also support images for the commands.
+- We support two lines of text for the commands.
+
+## Add a command
+
+In order to add a new command to the palette, you can include the command object in `useCommandsArrayWpcom` or create a new hook and include it to `useCommandPallette` hooks.
+
+## Command properties
+
+Every command has a set of properties that can be used to customize its behavior. The following properties are available:
+
+- `name`: Type `string`, used as identifier for the command. It's expected to be unique and `camelCase`. It's not visible for the user.
+- `label`: Type `string`, used as the label for the command. Visible in the command palette.
+- `subLabel`?: Type `string`, used as a second line for the command. Visible in the command palette.
+- `searchLabel`?: Type `string`, used to match the user search. It's not visible for the user. We recommend to use `_x()` to add context to the translators.
+- `callback`: Type `function`, used to execute the command when the user clicks on it or presses enter on that command.
+- `context`?: Type `string[]`, a list of URL paths for which the command will have priority when displayed.
+- `icon`?: Type `JSX.Element`, used to display an icon for the command. It's visible in the command palette.
+- `image`?: Type `JSX.Element`, used to display an image for the command, instead of an icon. It's visible in the command palette.
+- `siteFunctions`?: Type `object`, used for nested commands that need to execute a function when a site is selected in the command palette as a second step.
```

Comment:
Could clarify what happens if there's already active site (i.e. most pages outside `/sites`, `/read` and `/me`.

Reply:
I added the current behaviour here https://github.com/Automattic/wp-calypso/pull/84591/commits/4c9db3b0b165842c81d89950ba3dbf13c14e4d49

After trying the palette globally I see we can iterate on the behaviour, and for selected sites it could navigate directly between pages. My other idea for when a site is selected, we can display the current site as first option always.

Reply:
Additional site switcher upon commands feels like a lot of friction for what's supposed to be quick commands, if the site is already "selected". There's the quick-site switcher in the sidebar, after all.

Reply:
I agree. In the single site context, the command should apply to the current site and you shouldn't see the site switcher.